test/ExecutionFramework.cpp

```diff
@@ -170,6 +171,33 @@ void ExecutionFramework::sendMessage(bytes const& _data, bool _isCreation, u256
 		m_transactionSuccessful = (m_gas != m_gasUsed);
 }
 
+void ExecutionFramework::waitForTransaction(std::string const& _txHash) const
+{
+	int polls = 0;
+	while(true)
+	{
+		polls++;
+		auto pendingBlock = m_rpc.eth_getBlockByNumber("pending", false);
+
+		bool isEmpty = pendingBlock["transactions"].empty();
+
+		if (isEmpty && polls < 3000)
+		{
+			if (polls == 200)
+			{
+				cerr << "Note: Already used 200 iterations while waiting for transaction confirmation. Issuing an eth_flush request." << endl;
+				m_rpc.rpcCall("eth_flush");
+			}
+
+			continue;
+		}
+
+		BOOST_REQUIRE_MESSAGE(!isEmpty, "Failed to get transaction confirmation after 3000 iterations.");
```

Comment:
`BOOST_REQUIRE_*` will terminate the test, so the next line will not be reached.

Reply:
That's desired, in the next line we try to access the array index 0, which we can't do if it is empty

Reply:
Ah sorry, didn't read the full thing :)